### DISCUSSION TITLE
Multiple GUI - pretty essential, but kinda QoL - ideas

### DISCUSSION THREAD

**Discussion by arthurkallinen** [Shelf-nu/shelf.nu#639] [Ideas]
- Ability to rezise/hide left menu for lower resolution screens.
![brave_idlWhFvvf9](https://github.com/Shelf-nu/shelf.nu/assets/44576346/6294ad78-e069-44bf-b691-830de5d28c4c)
- Custom fields to be a sortable column in asset list view
![brave_zdbYD0cFhI](https://github.com/Shelf-nu/shelf.nu/assets/44576346/e10bf1bd-6bfb-492d-b47d-37a5561a655c)
- Reordering custom field items (for example, dropdown list items currently stuck in order of input. ++ for sort by name option.
![brave_FMyy3MweC7](https://github.com/Shelf-nu/shelf.nu/assets/44576346/6b3c71ec-4b0a-498e-93c9-445e4235d2f2)
- Scale to fit as default or option for thumbnail, if asset thumbnail is not square, it gets cutoff.
![brave_NQrMHxWEXE](https://github.com/Shelf-nu/shelf.nu/assets/44576346/6f4eead2-9dce-40e8-8ef9-54d7353f222d)
- Edit with double click on asset's page, or via at least 1 edit button (currently thru the menu, ideally button where in the next image there is Save button.
![brave_wvhioZ14VF](https://github.com/Shelf-nu/shelf.nu/assets/44576346/97c7f4ad-2e0e-4576-9549-c35d63edc282)
- Save button at top of page when editing. Even just for a rename you now have to scroll to bottom, which especially on mobile is long way with custom fields. Also, there is a lot of wasted space on the right of item on PC, so the custom labels / Details should be horizontally stacked instead of vertically.
![brave_i15NyeXe1F](https://github.com/Shelf-nu/shelf.nu/assets/44576346/06a9c674-f04c-4c70-87d3-06b11eb8cd8e)
- Autosaving asset, at least when creating it (Relating to upcoming: [Issue#640](https://github.com/Shelf-nu/shelf.nu/issues/640) )
- Move custom labels to details screen, not fast enough to look for on mobile
https://github.com/Shelf-nu/shelf.nu/assets/44576346/6fed039a-07e1-4f20-8bd1-39efdab136a1
- Ability to create categories/tags on asset creation/edit page by writing one onto the field and maybe there being to "Create" button if it doesn't exist 
// 
- Ability to make searching item/tag/category into creating one if not found. This would work similarly like you would add tags onto youtube video. Just write them down. The create pages are not necessary and creates extra steps for creating assets. (Yes there is the "create category" button, but it just redirects to blank category page, and clicking it will discard progress).

Make search box morph into tag name and Description and create buttons appear on the right
![brave_nYdML6eIzY](https://github.com/Shelf-nu/shelf.nu/assets/44576346/8c4da2ac-5553-476f-af29-fbc895e52b73)

- Clicking Category/Tag/Location/Custom would filter / sort by it rather than just opening the asset
![brave_xQOOemQdu9](https://github.com/Shelf-nu/shelf.nu/assets/44576346/bfa2edeb-f404-4b6e-8b37-7ddf6e7a1c7a)

**Comment by arthurkallinen** [comment]
@DonKoko you wanted a ping

**Comment by arthurkallinen** [comment]
- Display time of custody change with date
![brave_MNrdg5ayOL](https://github.com/Shelf-nu/shelf.nu/assets/44576346/8ecbd05a-1769-4c14-beed-b6ca48818441)

**Comment by DonKoko** [comment]
hey @arthurkallinen,
Thanks for opening a discussion and giving some suggestions. As a preface I would say we really appreciate it and there are some great things there that we would love to do, however have to consider the limitations of our time and the size of our team. Please find below my answers: 

> Ability to rezise/hide left menu for lower resolution screens.

This is a cool feature, but thats all that we see it as for now. We already discussed it and we wanted to build it but in the end we just decided to give the 2 options as it was faster to ship and at this point in time this is our highest prio. I see this as something we can do if we have some available time between features but not the highest prio. Feel free to create a feature request for this and we will add it to the backlog so we can make it work once we have availability.

> Custom fields to be a sortable column in asset list view
> Reordering custom field items (for example, dropdown list items currently stuck in order of input. ++ for sort by name option.

This is amazing and we have it planned, however it comes with quite some complexities and we have it blocked by some other issues we have to solve first. Feel free to create 1 feature request issue for this and we will pick it up as soon as we can or place a bounty once the blockers are solved.

> Scale to fit as default or option for thumbnail, if asset thumbnail is not square, it gets cutoff.

That is great feedback to have. We were not sure which approach is better and we never heard from users before you. Is this what you mean? The image will be shown in its full size but space around it will be empty.
![Screenshot 2024-01-08 at 15 19 06](https://github.com/Shelf-nu/shelf.nu/assets/7840007/cd674fef-fcc5-4888-bb43-237d5c14eadb)

> Edit with double click on asset's page, or via at least 1 edit button (currently thru the menu, ideally button where in the next image there is Save button.

We have put the edit button behind the actions dropdown on purpose as we see more value in having a clean UI rather than one with many buttons.

> Save button at top of page when editing. Even just for a rename you now have to scroll to bottom, which especially on mobile is long way with custom fields. Also, there is a lot of wasted space on the right of item on PC, so the custom labels / Details should be horizontally stacked instead of vertically.

We are currently improving the general structure of our UI and making some changes and this is one of the things planned and will be done. @hunar1312 could you create an issue for this and we can look at it together to decide how to do it technically, but the idea is to have the save button both on top and bottom on edit/new routes.

> Move custom labels to details screen, not fast enough to look for on mobile

This is considered in a complete redesign we are doing of the asset page and will be done somewhere in the near future. Thanks for the suggestion.


> Ability to make searching item/tag/category into creating one if not found. This would work similarly like you would add tags onto youtube video. Just write them down. The create pages are not necessary and creates extra steps for creating assets. (Yes there is the "create category" button, but it just redirects to blank category page, and clicking it will discard progress).

I like this suggestion. @jurrejansen can you think of how we can make something like this work that fits with our current UI? 

> Clicking Category/Tag/Location/Custom would filter / sort by it rather than just opening the asset

We are aware of this but it requires us to re-build some components that we use everywhere so we want to do it right but its planned already once its blockers are solved.

> Display time of custody change with date

I will look into this.

**Reply by DonKoko** [reply]
@arthurkallinen FYI, the time is added to the notes: https://github.com/Shelf-nu/shelf.nu/pull/646

**Comment by arthurkallinen** [comment]
Fully understand resources are limited. Still glad to hear that the suggestions are plausible. Especially the categry/tag/sorting would be huge. Good to have this efficient way to help with suggestions and relatively fast response - leaves a nice feeling of being heard.

> 
> > Ability to rezise/hide left menu for lower resolution screens.
> 
> This is a cool feature, but thats all that we see it as for now. We already discussed it and we wanted to build it but in the end we just decided to give the 2 options as it was faster to ship and at this point in time this is our highest prio. I see this as something we can do if we have some available time between features but not the highest prio. Feel free to create a feature request for this and we will add it to the backlog so we can make it work once we have availability.
> 
Alright, will create feature request.

> > Custom fields to be a sortable column in asset list view
> > Reordering custom field items (for example, dropdown list items currently stuck in order of input. ++ for sort by name option.
> 
> This is amazing and we have it planned, however it comes with quite some complexities and we have it blocked by some other issues we have to solve first. Feel free to create 1 feature request issue for this and we will pick it up as soon as we can or place a bounty once the blockers are solved.
> 
Sounds great. I'll create feature request.

> > Scale to fit as default or option for thumbnail, if asset thumbnail is not square, it gets cutoff.
> 
> That is great feedback to have. We were not sure which approach is better and we never heard from users before you. Is this what you mean? The image will be shown in its full size but space around it will be empty.
![Screenshot 2024-01-08 at 15 19 06](https://private-user-images.githubusercontent.com/7840007/294911112-cd674fef-fcc5-4888-bb43-237d5c14eadb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ3MzA0MjUsIm5iZiI6MTcwNDczMDEyNSwicGF0aCI6Ii83ODQwMDA3LzI5NDkxMTExMi1jZDY3NGZlZi1mY2M1LTQ4ODgtYmI0My0yMzdkNWMxNGVhZGIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMDhUMTYwODQ1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzdjN2VmMzJiNjJiMjc1MzY2MjcxNzdkZmM4MDEzNDdkYTQzZTdiZmM4ZjgyOWE2ZTgyYjdhM2IxODQ4Mzk3ZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.M3xyozXVFFTINwILE8wvshVq6zIOLjQr-3Cgp2KUy1s)
>
Yes. This would be the case. Understandably there is risk of the image being too small, but I think it's a smaller problem.

> > Edit with double click on asset's page, or via at least 1 edit button (currently thru the menu, ideally button where in the next image there is Save button.
> 
> We have put the edit button behind the actions dropdown on purpose as we see more value in having a clean UI rather than one with many buttons.
> 
I'm always function over visuals, but I understand that philosophy as well. I guess this goes under the UX redesign, but placing the menu button closer to center, instead of being in the right edge, as on PC it is quite far away from all action?

> > Save button at top of page when editing. Even just for a rename you now have to scroll to bottom, which especially on mobile is long way with custom fields. Also, there is a lot of wasted space on the right of item on PC, so the custom labels / Details should be horizontally stacked instead of vertically.
> 
> We are currently improving the general structure of our UI and making some changes and this is one of the things planned and will be done. @hunar1312 could you create an issue for this and we can look at it together to decide how to do it technically, but the idea is to have the save button both on top and bottom on edit/new routes.
> 
Glad to hear!

> > Ability to make searching item/tag/category into creating one if not found. This would work similarly like you would add tags onto youtube video. Just write them down. The create pages are not necessary and creates extra steps for creating assets. (Yes there is the "create category" button, but it just redirects to blank category page, and clicking it will discard progress).
> 
> I like this suggestion. @jurrejansen can you think of how we can make something like this work that fits with our current UI?
>
Would be nice!

> > Clicking Category/Tag/Location/Custom would filter / sort by it rather than just opening the asset
> 
> We are aware of this but it requires us to re-build some components that we use everywhere so we want to do it right but its planned already once its blockers are solved.
>
Good to hear!
